boot/boot_serial/src/boot_serial.c

```diff
@@ -683,12 +687,17 @@ boot_serial_start(const struct boot_uart_funcs *f)
     max_input = sizeof(in_buf);
 
     off = 0;
-    while (1) {
+    while (timeout_in_ms > 0 || bs_entry) {
         MCUBOOT_CPU_IDLE();
         MCUBOOT_WATCHDOG_FEED();
+#ifdef __ZEPHYR__
+        uint32_t start = k_uptime_get_32();
+#else
+        uint32_t start = os_get_uptime_usec()/1000;
+#endif
```

Comment:
I am little against adding more __ZEPHYR__/else blocks in common code: if number of supported systems would increase then we would have a lot of these.
Can we solve that by adding some function declaration that different systems could provide own implementations for?

Reply:
I understand what you mean, but not sure where to put the platform dependant header in that case.. i'm only familiar with zephyr, so for any other build systems/platforms I would need some advice to proceed. Could you agree if I would make it a define at top to start with, so the code is not double ifdef'ed? 
```
#ifdef __ZEPHYR__
#define MCUBOOT_UPTIME_MS_U32 (k_uptime_get_32())
#else
#define MCUBOOT_UPTIME_MS_U32 (os_get_uptime_usec()/1000)
#endif
```

Reply:
You do not have to do this for every system (look https://github.com/mcu-tools/mcuboot/pull/1280/files#r791572629), when you clearly state that that system does not support it.
You just have to propose some common function name like `os_time_uptime_ms` (or something) and provide implementation for systems that you know and make sure that these systems link that function.
Only systems that set `MCUBOOT_SERIAL_WAIT_FOR_DFU != 0` will have to implement the function, so if somebody would look into the mcuboot_config.h of own system then the person will know that this is hardcoded to 0 and some work is expected to happen before the timeout works.
Look into for example how each system code within mcuboot provides own `flash_area_id_to_multi_image_slot`.